### PR TITLE
SPARKC-362: Added support for INSERT ... IF NOT EXISTS

### PR DESCRIPTION
**Pull request by randomf**
Added support for IF NOT EXISTS behaviour when inserting data and specifying ifNotExists = true in WriteConf.

### CODE REVIEW

**Comment by RussellSpitzer** [comment]
test this please

File: spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/writer/TableWriter.scala [hunk 1]
```diff
@@ -34,21 +34,13 @@ class TableWriter[T] private (
   val columnNames = rowWriter.columnNames diff writeConf.optionPlaceholders
   val columns = columnNames.map(tableDef.columnByName)
 
-  val defaultTTL = writeConf.ttl match {
```
Comment (RussellSpitzer) [inline spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/writer/TableWriter.scala]: Why remove these variables? Are these dead code?

Comment (randomf) [inline spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/writer/TableWriter.scala]: Yes, these were unused and not referenced anywhere in the code.

Comment (RussellSpitzer) [inline spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/writer/TableWriter.scala]: Thanks for the cleanup, this all looks clean to me. Once the tests pass on our jenkins (those "continuous-integration" things below i'll merge

**Comment by RussellSpitzer** [comment]
retest this please

**Comment by RussellSpitzer** [comment]
For some reason 2.10 is just unhappy today, but it has nothing to do with the code. Merged Thanks @randomf for your contribution!

**Pull request merged**